packages/common/src/signHelpers.ts

```diff
@@ -55,37 +55,41 @@ export function signTransaction(
 
 /**
  * @param account the address of the account that will sign this message.
- * @param message the message to sign, assumed to be utf8 encoded.
+ * @param message the message to sign, assumed to be utf8 encoded string or a Uint8Array/buffer.
  */
-function getMessageDigest(account: AccountAddress, message: string): Buffer {
+function getMessageDigest(
+    account: AccountAddress,
+    message: string | Uint8Array
+): Buffer {
     const prepend = Buffer.alloc(8, 0);
-    const rawMessage = Buffer.from(message, 'utf8');
+    const rawMessage =
+        typeof message === 'string' ? Buffer.from(message, 'utf8') : message;
     return sha256([account.decodedAddress, prepend, rawMessage]);
 }
 
 /**
  * Helper function to sign a message.
- * Note that this function prepends the string "MyGoodPrepend" to ensure that the message is not a transaction.
+ * Note that this function prepends 8 zero-bytes to ensure that the message is not a transaction.
```

Comment:
I think it technically prepends the sender address (32 bytes) + 8 zero bytes, right?

Reply:
Yes, I corrected it. :sweat: